addon/mixins/scoped-query-cache.js

```diff
@@ -68,10 +73,38 @@ export default Ember.Mixin.create({
       assert('shouldCache needs to be a boolean value', typeof(shouldCache) === 'boolean');
 
       if (shouldCache) {
-        queriesForModel[cacheReadStringifiedQuery] = queryPromise;
+        this._scopedQueryCache.cache.add(modelName, cacheReadStringifiedQuery, queryPromise);
       }
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    if (this._scopedQueryCache.decayId) {
+      this.cancelCacheDecay();
+    }
+
+    this._scopedQueryCache.decayId = run.later(() => {
+      if (this.isDestroying || this.isDestroyed) {
+        return;
+      }
+
+      this._scopedQueryCache = this._createScopedQueryCache();
+
+      callback && typeof callback === 'function' && callback();
+    }, delay);
+  },
+
+  cancelCacheDecay() {
+    run.cancel(this._scopedQueryCache.decayId);
+    this._scopedQueryCache.decayId = null;
+  },
+
+  _createScopedQueryCache() {
```

Comment:
Ya I've already taken it part way there. It's likely going the full distance and moving the schedule in there seems the correct path.